### ISSUE TITLE
Transitive python_requires does no longer work since Conan 1.17 (in combination with lockfile)

### ISSUE THREAD

**Issue by Aalmann** [conan-io/conan#5529]
Hi @memsharded, @lasote and @jgsogo,

since Conan 1.17 there is an issue with transitive python_requires references.

To reproduce you just need the following:
1. define a simple conan file which can be used as python_requires. Let's call it "base_helper".
```python
from conans import ConanFile

def hello_world():
    message = "Hello World"
    print(message)
    return message

class MyBaseConan(ConanFile):
    name = "base_helper"
    version = "1.2.3"
    #  ..
```
2. create another simple conan file which can be used as python_requires. Let's call it "impl_helper"
```python
from conans import ConanFile, tools
from conans import python_requires #@UnresolvedImport

my_base= python_requires("base_helper/1.2.3@user/channel")
test = my_base.hello_world()

class ImplConan(ConanFile):
    name = "impl_helper"
    version = "2.3.4"
```
3. create a "normal" package recipe which python_requires "impl_helper"
```python
from conans import ConanFile, tools
from conans import python_requires #@UnresolvedImport

python_requires("impl_helper/2.3.4@user/channel")

class DummyexampleConan(ConanFile):
    name = "DummyExample"
    version = "0.0.1"
    settings = "os", "compiler", "build_type", "arch"
```

If you now call ```conan install DummyExample/0.0.1@user/channel``` and after that ```conan source``` (or something else) it will end in an error. Also ```conan graph lock DummyExample/0.0.1@test/testing``` shows the same result.

Best
Aalmann

**Comment by memsharded** [comment]
Thanks very much for the detailed reproducible case. I have been able to reproduce it and working on a fix, will try to release it soon.

**Comment by Aalmann** [comment]
Thanks for fixing it.

**Comment by memsharded** [comment]
Closed by #5531, will be in Conan 1.17.2
